edgeless_api/proto/messages.proto

```diff
@@ -45,21 +45,30 @@ message SpawnFunctionRequest {
     InstanceId instance_id = 1;
     // The function class specification.
     FunctionClassSpecification code = 2;
-    // Mapping of the outputs to concrete function/resource instances.
-    map<string, InstanceId> output_mapping = 3;
-    // The set of annotations associated to this function instance.
-    map<string, string> annotations = 5;
+    // The set of annotations associated with this function instance.
+    map<string, string> annotations = 3;
     // The state specification.
-    StateSpecification state_specification = 6;
+    StateSpecification state_specification = 4;
+}
+
+// Message to start a new resource instance.
+message StartResourceRequest {
+    // Resource type.
+    string class_type = 1;
+    // Configuration of the resource, which depends on the resource type.
+    map<string, string> configurations = 2;
 }
 
-// Response message to the creation a new function instance.
-message SpawnFunctionResponse {
+
+// Response message to the creation a new function instance or the start of a
+// resource.
+message StartComponentResponse {
```

Comment:
I am not sure I understand what you mean, let me guess.

This message is used as a response to the request to spawn a function instance or start a resource, i.e., in response to one of the following messages:

- SpawnFunctionRequest
- StartResourceRequest

For function instances, I kept the verb _spawn_, which seems more appropriate because a function instance is something that is put into action that did not exist before.

On the other hand, for resources, I used the verb _start_ since it is a request to trigger something from a resource provider, which may or may not require a new process/thread/whatever.

To capture the generic nature of the response, I used "Component" in the `StartComponentResponse`, but this unfortunately only refers to the start (resource) not the spawn (function).

Is this the naming inconsistency you identified?